services/src/main/java/org/keycloak/services/clientpolicy/executor/IntentClientBindCheckExecutor.java

```diff
@@ -0,0 +1,220 @@
+
+    public IntentClientBindCheckExecutor(KeycloakSession session) {
+        this.session = session;
+    }
+
+    @Override
+    public String getProviderId() {
+        return IntentClientBindCheckExecutorFactory.PROVIDER_ID;
+    }
+
+    @Override
+    public void setupConfiguration(IntentClientBindCheckExecutor.Configuration config) {
+        this.configuration = Optional.ofNullable(config).orElse(createDefaultConfiguration());
+    }
+
+    @Override
+    public Class<Configuration> getExecutorConfigurationClass() {
+        return Configuration.class;
+    }
+
+    public static class Configuration extends ClientPolicyExecutorConfigurationRepresentation {
+
+        @JsonProperty("intent-client-bind-check-endpoint")
+        protected String intentClientBindCheckEndpoint;
+
+        @JsonProperty("intent-name")
+        protected String intentName;
+
+        public String getIntentClientBindCheckEndpoint() {
+            return intentClientBindCheckEndpoint;
+        }
+
+        public void setIntentClientBindCheckEndpoint(String intentClientBindCheckEndpoint) {
+            this.intentClientBindCheckEndpoint = intentClientBindCheckEndpoint;
+        }
+
+        public String getIntentName() {
+            return intentName;
+        }
+
+        public void setIntentName(String intentName) {
+            this.intentName = intentName;
+        }
+    }
+
+    @Override
+    public void executeOnEvent(ClientPolicyContext context) throws ClientPolicyException {
+        switch (context.getEvent()) {
+        case AUTHORIZATION_REQUEST:
+            checkIntentClientBind((AuthorizationRequestContext)context);
+            break;
+        default:
+            return;
+        }
+    }
+
+    private Configuration createDefaultConfiguration() {
+        Configuration conf = new Configuration();
+        return conf;
+    }
+
+    private void checkIntentClientBind(AuthorizationRequestContext context) throws ClientPolicyException {
+        if (!isValidIntentClientBindCheckEndpoint()) {
+            throw new ClientPolicyException(OAuthErrorException.SERVER_ERROR, "invalid Intent Client Bind Check Endpoint setting");
+        }
+        ClientModel client = session.getContext().getClient();
+        String clientId = client.getClientId();
+        String intentId = retrieveIntentId(context.getAuthorizationEndpointRequest());
+        IntentBindCheckRequest request = new IntentBindCheckRequest();
+        request.setClientId(clientId);
+        request.setIntentId(intentId);
+        SimpleHttp simpleHttp = SimpleHttp.doPost(configuration.getIntentClientBindCheckEndpoint(), session)
+                .header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON)
+                .json(request);
+        IntentBindCheckResponse response = null;
+        try {
+            response = simpleHttp.asJson(IntentBindCheckResponse.class);
+        } catch (IOException e) {
+            logger.warnv("HTTP connection failure: {0}", e.getMessage());
```

Comment:
Is it possible to rather log whole stacktrace instead of just message? So also include `e` as second parameter to `warnv`? Again, it can be useful for debugging purposes (Exception message is good, but may not be that sufficient for all cases (especially if the real cause is some nested exception etc).

Reply:
Yes, I will log `e` itself instead of `e.getMessage()`.